include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -677,6 +659,13 @@ __parallel_transform_scan_single_group(oneapi::dpl::__internal::__device_backend
     // Specialization for devices that have a max work-group size of 1024
     constexpr ::std::uint16_t __targeted_wg_size = 1024;
 
+    using _ValueType = typename _InitType::__value_type;
+
+    // Although we do not actually need result storage in this case, we need to construct
```

Comment:
I have a prototype branch `dev/julianmi/scratch_storage` that supports cases where we don't need a result but want to keep the same return type. Would this be applicable here?

Reply:
Yes, I believe this is what we would like here. I've not looked in full detail at the branch, but on first glance it looks like what we would want.

Reply:
Ok, I'll productize it then and open a PR.

Reply:
Thanks!  We will review it and rebase upon it once its merged